persistqueue/queue.py

```diff
@@ -68,13 +68,24 @@ def __init__(self, path, maxsize=0, chunksize=100, tempdir=None):
         The tempdir parameter indicates where temporary files should be stored.
         The tempdir has to be located on the same disk as the enqueued data in
         order to obtain atomic operations.
+
+        The serialize parameter controls how enqueued data is serialized. It
+        must be a callable that accepts two args. The first arg is the data
+        to serialize, and the second arg is a byte mode file object the
+        serialization should be written to. It must serialize data so that
+        deserialize can read one object from the file at a time.
+
+        The deserialize parameter controls how enqueued data is deserialized.
+        It must be a callable that accepts one arg, a byte mode file object,
+        and reads one object that is returned.
```

Comment:
I went to implement msgpack and json serializers, and found that it was much easier to configure with callables than a single `serializer` with `load` and `dump` methods. let me know if you would prefer a different implementation.

Reply:
I have to say, I prefer the latter one, it is following the OOP, and easier to extend.
And do keep the backward compatibility.

Thanks
Peter

Reply:
okay. I'll restructure it to pass a serializer again on Monday (PDT timezone).

Reply:
👍